webrender/src/prim_store.rs

```diff
@@ -381,7 +385,10 @@ impl GradientData {
     // Compute an entry index based on a gradient stop offset.
     #[inline]
     fn get_index(offset: f32) -> usize {
-        (offset.max(0.0).min(1.0) * GRADIENT_DATA_RESOLUTION as f32).round() as usize
+        let clamped = offset.max(0.0).min(1.0);
+        let scaled = clamped * (GRADIENT_DATA_LAST - GRADIENT_DATA_FIRST) as f32 + GRADIENT_DATA_FIRST as f32;
```

Comment:
So this function returns indices between the `GRADIENT_DATA_FIRST` and `GRADIENT_DATA_LAST`, inclusive. It seems a little inconsistent that we have a border element to the left of `FIRST` (under index 0), but no border to the right of `LAST`.